metadata-ingestion/src/datahub/ingestion/source/vertexai.py

```diff
@@ -0,0 +1,497 @@
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.api.source import Source, SourceCapability, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata._schema_classes import (
+    AuditStampClass,
+    DataProcessInstanceInputClass,
+    DataProcessInstancePropertiesClass,
+    DatasetPropertiesClass,
+    TimeStampClass,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.dataset import DatasetProperties
+from datahub.metadata.schema_classes import (
+    MLModelDeploymentPropertiesClass,
+    MLModelGroupPropertiesClass,
+    MLModelPropertiesClass,
+    VersionTagClass,
+    _Aspect,
+)
+
+T = TypeVar("T")
+
+logger = logging.getLogger(__name__)
+
+class VertexAIConfig(EnvConfigMixin):
+    project_id: str = Field(description=("Project ID in Google Cloud Platform"))
+    region: str = Field(
+        description=("Region of your project in Google Cloud Platform"),
+    )
+    bucket_uri: Optional[str] = Field(
+        default=None,
+        description=("Bucket URI used in your project"),
+    )
+
+    vertexai_url: Optional[str] = Field(
+        default="https://console.cloud.google.com/vertex-ai",
+        description=("VertexUI URI"),
+    )
+
+    model_name_separator: str = Field(
+        default="_",
+        description="A string which separates model name from its version (e.g. model_1 or model-1)",
+    )
+
+
+@platform_name("vertexai")
+@config_class(VertexAIConfig)
+@support_status(SupportStatus.TESTING)
+@capability(
+    SourceCapability.DESCRIPTIONS,
+    "Extract descriptions for vertexai Registered Models and Model Versions",
+)
+@capability(SourceCapability.TAGS, "Extract tags for VertexAI Registered Model Stages")
+class VertexAISource(Source):
+    platform = "vertexai"
+
+    def __init__(self, ctx: PipelineContext, config: VertexAIConfig):
+        super().__init__(ctx)
+        self.config = config
+        self.report = SourceReport()
+        aiplatform.init(project=config.project_id, location=config.region)
+        self.client = aiplatform
+        self.endpoints = None
+
+    def get_report(self) -> SourceReport:
+        return self.report
+
+    def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
+        """
+        Main Function to fetch and yields work units for various VertexAI resources.
+        - Models and Model Versions from the Model Registry
+        - Training Jobs
+        """
+        # Fetch Models, Model Versions a from Model Registry
+        yield from self._get_ml_model_workunits()
+        # Fetch Training Jobs
+        yield from self._get_training_job_workunit()
+        # TODO Fetch Experiments and Experiment Runs
+
+    def _create_workunit(self, urn: str, aspect: _Aspect) -> MetadataWorkUnit:
+        """
+        Utility to create an MCP workunit.
+        """
+        return MetadataChangeProposalWrapper(
+            entityUrn=urn,
+            aspect=aspect,
+        ).as_workunit()
```

Comment:
use `auto_workunit(stream of mcps) -> stream of MWU` - already exists

move towards this structure

```
def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
	yield from auto_workunit(self.get_metadata())

def get_metadata() -> MetadataChangeProposalWrapper:
	...
```